app/src/main/java/org/mozilla/rocket/privately/PrivateModeActivity.kt

```diff
@@ -25,34 +28,37 @@ class PrivateModeActivity : LocaleAwareAppCompatActivity(), FragmentListener {
         // do not overwrite existing value
         visibility = visibility or (View.SYSTEM_UI_FLAG_LAYOUT_STABLE or View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN)
         window.decorView.systemUiVisibility = visibility
-
-        showHomeScreen()
     }
 
-    override fun onBackPressed() {
-        super.onBackPressed()
-        pushToBack()
-    }
+    @Override
+    override fun onSupportNavigateUp() = getNavController().navigateUp()
 
     override fun applyLocale() {}
 
     override fun onNotified(from: Fragment, type: FragmentListener.TYPE, payload: Any?) {
         when (type) {
-            FragmentListener.TYPE.TOGGLE_PRIVATE_MODE -> pushToBack()
+            TYPE.TOGGLE_PRIVATE_MODE -> pushToBack()
+            TYPE.SHOW_URL_INPUT -> showUrlInput(payload)
+            TYPE.DISMISS_URL_INPUT -> dismissUrlInput()
+            TYPE.OPEN_URL_IN_CURRENT_TAB -> openUrl(payload)
+            TYPE.OPEN_URL_IN_NEW_TAB -> openUrl(payload)
             else -> {
             }
         }
     }
 
-    private fun showHomeScreen() {
-        val fragment = this.createHomeFragment()
-        this.supportFragmentManager.beginTransaction()
-                .add(R.id.container, fragment, PrivateHomeFragment.FRAGMENT_TAG)
-                .commit()
-    }
-
-    private fun createHomeFragment(): PrivateHomeFragment {
-        return PrivateHomeFragment.create()
+    override fun onBackPressed() {
+        val cnt = supportFragmentManager.backStackEntryCount
```

Comment:
maybe add a comment about "Navigation can't work with backstack. if we have some manually added backstack, we must handle it before we forward the back key event to Navigation